File: perl/euler6.pl
#!/usr/bin/perl
# Project Euler #6
# Find the difference between the sum of the squares of the first one
#   hundred natural numbers and the square of the sum.
# Answer: 25164150


my $squareofsums = 0;
my $sumofsquares = 0;
foreach (1..100)
{
	$squareofsums += $_;
	$sumofsquares += $_*$_;
}
$squareofsums *= $squareofsums;

print $squareofsums - $sumofsquares, "\n";

File: java/Euler6.java
/* Project Euler #6
 * Find the difference between the sum of the squares of the first one
 *   hundred natural numbers and the square of the sum.
 * Answer: 25164150
*/

public class Euler6 {
	public static void main(String[] args) {
		long squareofsums = 0L;
		long sumofsquares = 0L;
		for (int i = 1; i < 101; i++)
		{
			squareofsums += i;
			sumofsquares += i*i;
		}
		squareofsums *= squareofsums;
		System.out.println(squareofsums - sumofsquares);
	}
}